File: ruby-glmaze/gamestate.rb
class GameState
  attr_reader :engine

  def initialize(engine)
    @engine = engine
  end

  def start
    nil
  end

  def stop
    nil
  end

  def pause
    nil
  end

  def resume
    nil
  end

  def render
    nil
  end

  def update
    nil
  end

  def keyDown(event)
    nil
  end
end

File: cpp-glmaze/src/gamestate.cpp
#include "gamestate.h"

GameState::GameState(Engine* e) : engine(e) { }
GameState::~GameState() { }
void GameState::Init() { }
void GameState::Term() { }
void GameState::Pause() { }
void GameState::Resume() { }
void GameState::Render() { }
void GameState::Update() { }
void GameState::OnKeyDown(SDL_KeyboardEvent *e) { }
